components/script/dom/mediasession.rs

```diff
@@ -194,6 +197,64 @@ impl MediaSessionMethods for MediaSession {
             None => self.action_handlers.borrow_mut().remove(&action.into()),
         };
     }
+
+    /// https://w3c.github.io/mediasession/#dom-mediasession-setpositionstate
+    fn SetPositionState(&self, state: &MediaPositionState) -> Fallible<()> {
+        // If the state is an empty dictionary then clear the position state.
+        if state.duration.is_none() && state.position.is_none() && state.playbackRate.is_none() {
+            if let Some(media_instance) = self.media_instance.get() {
+                media_instance.reset();
+            }
+            return Ok(());
+        }
+
+        // If the duration is not present or its value is null, throw a TypeError.
+        if state.duration.is_none() {
+            return Err(Error::Type(
+                "duration is not preset or its value is null".to_owned(),
```

Comment:
typo: present

Reply:
Oops. This is also typo..., Sorry.
done.